Added content for stacks and queues (#68)
WomenWhoCode/wwcsf-algos: topics/stacks-queues.md

# Stacks and Queues

## Go to:
 * [Resources](#resources)
 * [Problem Sets](#problem-sets)

___

## Resources
- [CMU Introduction to Data Structures (15121)](https://www.andrew.cmu.edu/course/15-121/lectures/Stacks%20and%20Queues/Stacks%20and%20Queues.html#:~:text=A%20queue%20is%20a%20container,%2Dout%20(FIFO)%20principle.&text=The%20difference%20between%20stacks%20and,item%20the%20least%20recently%20added.)
- Gayle Laakmaan McDowell + Hackerrank videos on stacks and queues:
    - [Stacks & Queues](https://www.youtube.com/watch?v=wjI1WNcIntg)
- [Blog by Educative](https://www.educative.io/blog/data-structures-stack-queue-java-tutorial)
- [Princeton University - Introduction to Stacks](https://introcs.cs.princeton.edu/java/43stack/) 

## Problem Sets

### Easy problems
Problem Statement | Solution | WWC example
--- | --- | ---
[Valid Parentheses](https://leetcode.com/problems/valid-parentheses/) | [Solution link](https://leetcode.com/problems/valid-parentheses/solution/) | please contribute your solution to our [code examples!](https://github.com/WomenWhoCode/wwcsf-algos/tree/master/code-examples/)

> [Min Stack](https://leetcode.com/problems/min-stack/) | [Solution discussion](https://leetcode.com/problems/min-stack/solution/) | please contribute your solution to our [code examples!](https://github.com/WomenWhoCode/wwcsf-algos/tree/master/code-examples/)

Reviewer: It looks like the solution for Min Stack is behind the paywall. I suggest directing to the solution discussion instead:
```suggestion
[Min Stack](https://leetcode.com/problems/min-stack/) | [Solution discussion](https://leetcode.com/problems/min-stack/discuss/?currentPage=1&orderBy=hot&query=) | please contribute your solution to our [code examples!](https://github.com/WomenWhoCode/wwcsf-algos/tree/master/code-examples/)
```

### Medium problems
Problem Statement | Solution | WWC example
--- | --- | ---

> [Binary Tree zig zag level order tree traversal ](https://leetcode.com/problems/binary-tree-zigzag-level-order-traversal/) | [Solution discussion](https://leetcode.com/problems/binary-tree-zigzag-level-order-traversal/solution/) | please contribute your solution to our [code examples!](https://github.com/WomenWhoCode/wwcsf-algos/tree/master/code-examples/)

Reviewer: Same thing here, solution behind the paywall. 
```suggestion
[Binary Tree zig zag level order tree traversal ](https://leetcode.com/problems/binary-tree-zigzag-level-order-traversal/) | [Solution discussion](https://leetcode.com/problems/binary-tree-zigzag-level-order-traversal/discuss/?currentPage=1&orderBy=hot&query=) | please contribute your solution to our [code examples!](https://github.com/WomenWhoCode/wwcsf-algos/tree/master/code-examples/)
```

> [Binary Tree inorder traversal](https://leetcode.com/problems/binary-tree-inorder-traversal/) | [Solution discussion](https://leetcode.com/problems/binary-tree-inorder-traversal/solution/) | please contribute your solution to our [code examples!](https://github.com/WomenWhoCode/wwcsf-algos/tree/master/code-examples/)

Reviewer: I'm sure no one will notice except for me, but let's label the solution 'solution link' to let people know it's the Solution and not the Discussion :)

```suggestion
[Binary Tree inorder traversal](https://leetcode.com/problems/binary-tree-inorder-traversal/) | [Solution link](https://leetcode.com/problems/binary-tree-inorder-traversal/solution/) | please contribute your solution to our [code examples!](https://github.com/WomenWhoCode/wwcsf-algos/tree/master/code-examples/)
```

> [Basic Calculator - II](https://leetcode.com/problems/basic-calculator-ii/) | [Solution discussion](https://leetcode.com/problems/basic-calculator-ii/solution/) | please contribute your solution to our [code examples!](https://github.com/WomenWhoCode/wwcsf-algos/tree/master/code-examples/)

Reviewer: Same thing here
```suggestion
[Basic Calculator - II](https://leetcode.com/problems/basic-calculator-ii/) | [Solution link](https://leetcode.com/problems/basic-calculator-ii/solution/) | please contribute your solution to our [code examples!](https://github.com/WomenWhoCode/wwcsf-algos/tree/master/code-examples/)
```

### Hard problems
Problem Statement | Solution | WWC example
--- | --- | ---

> [Trapping rain water](https://leetcode.com/problems/trapping-rain-water/) | [Solution discussion](https://leetcode.com/problems/kth-smallest-element-in-a-sorted-matrix/discuss/?currentPage=1&orderBy=most_votes&query=) | please contribute your solution to our [code examples!](https://github.com/WomenWhoCode/wwcsf-algos/tree/master/code-examples/)

Reviewer: This one is an error, Trapping Rain Water is missing the solution link
```suggestion
[Trapping rain water](https://leetcode.com/problems/trapping-rain-water/) | [Solution link](https://leetcode.com/problems/trapping-rain-water/solution/) | please contribute your solution to our [code examples!](https://github.com/WomenWhoCode/wwcsf-algos/tree/master/code-examples/)
```

> [Largest rectangle in histogram](https://leetcode.com/problems/largest-rectangle-in-histogram/) | [Solution discussion](https://leetcode.com/problems/wiggle-sort-ii/discuss/?currentPage=1&orderBy=most_votes&query=) | please contribute your solution to our [code examples!](https://github.com/WomenWhoCode/wwcsf-algos/tree/master/code-examples/)

Reviewer: Same thing here, missing solution link
```suggestion
[Largest rectangle in histogram](https://leetcode.com/problems/largest-rectangle-in-histogram/) | [Solution discussion](https://leetcode.com/problems/largest-rectangle-in-histogram/discuss/?currentPage=1&orderBy=hot&query=) | please contribute your solution to our [code examples!](https://github.com/WomenWhoCode/wwcsf-algos/tree/master/code-examples/)
```
